Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp

```diff
@@ -1285,6 +1408,28 @@ class ObjectAllocationSinkingPhase : public Phase {
         m_heap.assertIsValid();
     }
 
+    // This function removes unnecessary CheckInBounds nodes while lowering array allocations.
+    // It is safe to miss some CheckInBounds removals because their dependency, GetArrayLength,
+    // will always be replaced with a constant value.
+    void removeCheckInBoundsIfNeeded(Node* node)
+    {
+        switch (node->op()) {
+        case GetByVal:
+        case PutByVal:
+            for (unsigned i = 0; i < node->numChildren(); ++i) {
+                Node* child = m_graph.varArgChild(node, i).node();
+                if (child->op() == CheckInBounds) {
+                    child->decRef();
```

Comment:
I wonder if this ref count is still intact or not in this phase.

Reply:
For `NewArrayWithConstantSize` related nodes, I think so. The worst case is to recompute the ref count just before  `handleNode` with `enableLengthMaterialization = true`.

Reply:
Updated with explicit `m_graph.computeRefCounts()`.

Reply:
Removed `removeArrayDependentNodesIfNeeded `.